code/datums/flock/flock.dm

```diff
@@ -743,9 +761,15 @@ var/flock_signal_unleashed = FALSE
 						break
 				if (istype(O, /obj/machinery/light/lamp) || istype(O, /obj/machinery/computer3/generic/personal) || istype(O, /obj/machinery/computer3/luggable))
 					break
+				if (istype(O, /obj/machinery/vending) || istype(O, /obj/machinery/manufacturer) || istype(O, /obj/machinery/dispenser) || istype(O, /obj/machinery/disposal_pipedispenser) || istype(O, /obj/machinery/portable_atmospherics/canister) || istype(O, /obj/machinery/atmospherics/pipe/tank) || istype(O, /obj/machinery/portable_atmospherics/scrubber) || istype(O, /obj/machinery/chem_dispenser) || istype(O, /obj/machinery/chemicompiler_stationary))
+					params = O
+			else if (istype(O, /obj/submachine/seed_vendor))
+				params = O
+			else if (istype(O, /obj/reagent_dispensers) && !(istype(O, /obj/reagent_dispensers/cleanable) || istype(O, /obj/reagent_dispensers/still)))
+				params = O
 			var/dir = O.dir
 			var/replacementPath = flock_conversion_paths[keyPath]
-			var/obj/converted = new replacementPath(T)
+			var/obj/converted = new replacementPath(T, null, params)
```

Comment:
Since we don't know what type `replacementPath` is going to be, passing it `New` arguments for a specific type seems like a bad idea as we have no idea how the type will interpret those arguments.

Reply:
All things possible for replacementPath use at most 2 parameters in New(), this would just be an arbitrary third type that isn't used except when needed I think

Reply:
Yeah, that feels terrible. Probably better to just do a proc that instantiates a replacement path, and sets a var on converted flock objects of what the original object was.

Reply:
Setting a var wouldn't work since it's supposed to be getting current variables of something that exists, only other thing I could think of would be maintaining a list of all variables to their values of the converted object

Reply:
I don't see why that wouldn't work, but if you're really set on passing it as a parameter then put it as `atom/converted_from` in the flock structure parent `New` and just keep the fabricator as the only subtype that uses it. That way we're not committing too much OOP crime.